### ISSUE TITLE
Request for examples: Seq2seq

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/addons#335]
I believe there are many seq2seq examples already published, but it makes sense that we at least have one or two in our examples section for user experience.

Related:
https://google.github.io/seq2seq/
https://github.com/tensorflow/addons/blob/master/examples/README.md

**Comment by kazemnejad** [comment]
I can pick this up if it is ok.

Some ideas to start the example based on:
- For machine translation:
  - On toy English-to-German dataset (I think German-to-English has better user experience since we can demonstrate the outputs better) from http://opennmt.net/OpenNMT-tf/quickstart.html which contains 10k tokenized sentences.
  - Trivial seq2seq LSTM architecture with Attention Mechanism 
  - Use beam decoding to generate outputs
  - Maybe BLEU calculation

**Comment by seanpmorgan** [comment]
> I can pick this up if it is ok.
> 
> Some ideas to start the example:
> 
> * For machine translation:
>   
>   * On toy English-to-German dataset (I think German-to-English has better user experience since we can demonstrate the outputs better) from http://opennmt.net/OpenNMT-tf/quickstart.html which contains 10k tokenized sentences.
>   * Trivial seq2seq LSTM architecture with Attention Mechanism
>   * Use beam decoding to generate outputs
>   * Maybe BLEU calculation

That'd be great! Thanks and welcome to Addons!

**Comment by PyExtreme** [comment]
Hi @seanpmorgan @kazemnejad , Is it still an  _Open Issue_? 
Asking because I am interested to work on it.

**Comment by kazemnejad** [comment]
Hi @PyExtreme, Thank you for showing your interest. 
Yes, actually I'm currently working on it. I was waiting for the #375 to be fixed and it got fixed 3 days ago in #503. I think I can submit a PR in the next few days. Thus you can work on that PR if you want.

**Comment by matthen** [comment]
+1 any progress on this? 
I'd appreciate an example of how to create and train a keras decoder with attention

**Comment by kazemnejad** [comment]
Hi @Mainak431,  Sorry for the inconvenience.

Actually, the draft of this example is present at [my fork](https://github.com/kazemnejad/addons/tree/seq2seq_example_draft/docs/tutorials). However, in the default Keras training mode (graph mode) + tf.data.dataset (where the default mode is eager), there is a bug related to caching of tensor dimension which I guess is from the `AttentionWrapper`. Unfortunately, I'm busy with my university's works and I could not work on that bug, so I would be happy if someone could find the source of that bug and help the progress.

**Comment by matthen** [comment]
thanks for sharing @kazemnejad! Where do you get the bug? I was getting an error at some point from the rnn cell being wrapped by `AttentionWrapper` that it was being passed a rank 1 tensor when it was expecting a rank 2

**Comment by zhedongzheng** [comment]
Thanks @kazemnejad  for submitting the fix. Based on his contribution, I have successfully get the seq2seq code running for a semantic parsing task.

For anyone who is interested, the notebook is [here](https://nbviewer.jupyter.org/github/zhedongzheng/tensorflow-nlp/blob/master/finch/tensorflow2/semantic_parsing/tree_slu/main/lstm_seq2seq_tf_addons.ipynb)

**Comment by John-8704** [comment]
+1 Any progress on this?

I'd really appreciate if someone could provide an example on how to build seq2seq NMT model with attention and beam search wrappers.  I couldn't find any examples.

**Comment by seanpmorgan** [comment]
cc @Om-Pandey to see if this is an issue you would like to be assigned.

**Comment by Om-Pandey** [comment]
Yeah sure.. I would love to solve this.

**Comment by seanpmorgan** [comment]
> Yeah sure.. I would love to solve this.

Great let us know if you have any questions regarding [adding a tutorial](https://github.com/tensorflow/addons/blob/master/docs/tutorials/README.md)!

**Comment by kazemnejad** [comment]
@Om-Pandey I am glad that you accepted this. I will be more than happy to give more information about the bugs that I was referring to. Please feel free to contact me if it is needed.

**Comment by Om-Pandey** [comment]
@kazemnejad thank you so much... help is much needed 😅. @seanpmorgan based on earlier conversations in this thread, just wanted to clarify something, wouldn't it be better if we included a simple seq-2-seq NMT tutorial and gave separate methods and necessary explanation for attention modeling and beam/lexicon search which the reader can include if needed, rather than hard coding it into the structure in one workflow ?

**Comment by seanpmorgan** [comment]
> @kazemnejad thank you so much... help is much needed . @seanpmorgan based on earlier conversations in this thread, just wanted to clarify something, wouldn't it be better if we included the trivial seq-2-seq NMT tutorial and gave separate methods and necessary explanation for attention modeling and beam/lexicon search which the reader can include if needed, rather than hard coding it into the structure in one workflow ?

Yes, we're more than willing to have several different seq2seq tutorials

**Comment by Om-Pandey** [comment]
Hey @seanpmorgan , please check #806 and merge, Thanks!
